src/tool_sleep.c

```diff
@@ -51,7 +51,7 @@ void tool_go_sleep(long ms)
   struct timeval timeout;
   timeout.tv_sec = ms / 1000L;
   ms = ms % 1000L;
-  timeout.tv_usec = ms * 1000L;
+  timeout.tv_usec = (int)ms * 1000L;
```

Comment:
you're still multiplying it by a long constant.. you don't get a warning with that?